### ISSUE TITLE
Uncouple UI elements from DraculaComment

### ISSUE THREAD

**Issue by benknoble** [dracula/vim#181]
This is mostly an RFC for @dsifford, since he put the code together I'm talking about.

# TL;DR

I want to change a couple of lines like

```
hi! link LineNr       DraculaComment
```

to

```
call s:h('LineNr', s:comment)
```

The primary reason is to allow users to customize Comment styles (e.g., `cterm=italic`) without affecting UI components.

What do you think? Could we do this? I think it's minimal disruption, but I don't know what other effects it will have for maintainability down the road—the downside I see is if we update `DraculaComment` but not `LineNr`, which seems unlikely given how stable things are right now.

## Background

I've been using [Victor Mono](https://rubjo.github.io/victor-mono/) as my main font for a while, and I've grown to like its italic support (they're cursive in the standard variant, but there is an oblique variant too).

The author of the font likes these cursive italics in comments, as he feels it gives them a human touch.

I would like to try this out, but doing `hi DraculaComment cterm=italic gui=italic` also affects `LineNr` and `SignColumn`. I don't use the sign column much, but it is annoying to have the line numbers go italic.

**Comment by dsifford** [comment]
Sounds like a reasonable change to me. Go for it @benknoble! 👍
